### PR TITLE
feat: Client decoupling of Connection and Netty

### PR DESCRIPTION
**Pull request by conghuhu**
## What is the purpose of the change

Decouple Connection from remoting api to netty, and sink it into specific implementation.  

image

## Brief changelog
- new org.apache.dubbo.remoting.api.connection.ConnectionHandler
- new org.apache.dubbo.remoting.api.connection.AbstractConnectionClient extends AbstractClient
- new org.apache.dubbo.remoting.transport.netty4.NettyConnectionClient extends AbstractConnectionClient
- new org.apache.dubbo.remoting.transport.netty4.NettyConnectionHandler implements ConnectionHandler

## Verifying this change

- org.apache.dubbo.remoting.transport.netty4.ConnectionTest
- org.apache.dubbo.rpc.protocol.tri.TripleProtocolTest
- org.apache.dubbo.rpc.protocol.tri.TripleInvokerTest

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Review by AlbumenJ — APPROVED** [review]

**Review by AlbumenJ — COMMENTED** [review]
@guohao 
@EarthChen 
PTAL

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp2Protocol.java [hunk 1]
```diff
@@ -173,8 +178,12 @@ public void configClientPipeline(URL url, ChannelPipeline pipeline, SslContext s
                     DEFAULT_MAX_HEADER_LIST_SIZE)))
             .frameLogger(CLIENT_LOGGER)
             .build();
-        final Http2MultiplexHandler handler = new Http2MultiplexHandler(
-            new TripleClientHandler(frameworkModel));
-        pipeline.addLast(codec, handler, new TripleTailHandler());
+        final Http2MultiplexHandler handler = new Http2MultiplexHandler(new SimpleChannelInboundHandler<Object>() {
+            @Override
+            protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
+                logger.debug("triple client get msg from Http2MultiplexHandler: " + msg);
```
Comment (CrazyHZM) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp2Protocol.java:184]: Remove unused log

File: dubbo-remoting/dubbo-remoting-netty/src/main/java/org/apache/dubbo/remoting/transport/netty/NettyPortUnificationTransporter.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public AbstractPortUnificationServer bind(URL url, ChannelHandler handler) throw
     }
 
     @Override
-    public Client connect(URL url, ChannelHandler handler) throws RemotingException {
+    public AbstractConnectionClient connect(URL url, ChannelHandler handler) throws RemotingException {
         return null;
     }
```
Comment (CrazyHZM) [inline dubbo-remoting/dubbo-remoting-netty/src/main/java/org/apache/dubbo/remoting/transport/netty/NettyPortUnificationTransporter.java:38]: Behavior should be consistent with dubbo-remoting/dubbo-remoting-netty/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyPortUnificationTransporter.java

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

File: dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp2Protocol.java [hunk 3]
```diff
@@ -173,8 +178,12 @@ public void configClientPipeline(URL url, ChannelPipeline pipeline, SslContext s
...
-        final Http2MultiplexHandler handler = new Http2MultiplexHandler(
-            new TripleClientHandler(frameworkModel));
-        pipeline.addLast(codec, handler, new TripleTailHandler());
+        final Http2MultiplexHandler handler = new Http2MultiplexHandler(new SimpleChannelInboundHandler<Object>() {
+            @Override
+            protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
+                // Do nothing.
+            }
+        });
+        pipeline.addLast(codec, handler, new TripleClientHandler(frameworkModel), new TripleTailHandler());
```
Comment (CrazyHZM) [inline dubbo-rpc/dubbo-rpc-triple/src/main/java/org/apache/dubbo/rpc/protocol/tri/TripleHttp2Protocol.java:187]: If there is no special reason, this logic does not need to be changed.

**Review by CrazyHZM — CHANGES_REQUESTED** [review]

**Review by CrazyHZM — APPROVED** [review]
LGTM

File: dubbo-remoting/dubbo-remoting-netty/src/main/java/org/apache/dubbo/remoting/transport/netty/NettyConnectionClient.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    @Override
+    public void onConnected(Object channel) {
+    }
+
+    @Override
+    public void onGoaway(Object channel) {
+    }
+
+    @Override
```
Comment (AlbumenJ) [inline dubbo-remoting/dubbo-remoting-netty/src/main/java/org/apache/dubbo/remoting/transport/netty/NettyConnectionClient.java:57]: Shall we throw exception if users use netty3?

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**